libcontainer/cgroups/fs2/create.go

```diff
@@ -95,10 +96,20 @@ func CreateCgroupPath(path string, c *configs.Cgroup) (Err error) {
 					}
 				}()
 			}
-			// Write cgroup.type explicitly.
-			// Otherwise ENOTSUP may happen.
-			cgType := filepath.Join(current, "cgroup.type")
-			_ = ioutil.WriteFile(cgType, []byte("threaded"), 0644)
+			// There are 4 types: 'domain', 'threaded', 'domain threaded', and 'domain invalid'
+			// If got 'domain invalid', we should check parent's cgroup type.
+			// When no domain controller is enabled, we can write "threaded" without retuning erro
+			cgTypeFile := filepath.Join(current, "cgroup.type")
+			cgType, _ := ioutil.ReadFile(cgTypeFile)
+			if strings.TrimSpace(string(cgType)) == "domain invalid" {
+				cgTypeParentFile := filepath.Join(current, "../cgroup.type")
+				cgTypeParent, _ := ioutil.ReadFile(cgTypeParentFile)
+				if bytes.HasPrefix(cgTypeParent, []byte("domain")) {
```

Comment:
IIUC we don't need to check parent, we just need to check whether the current config contains domain controller